File: AndroidStudioProjects/DaggerDemoJava/app/src/main/java/com/example/myapplication/PetrolEngine.java
package com.example.myapplication;

import android.util.Log;

import javax.inject.Inject;

public class PetrolEngine implements Engine{

    @Inject
    public PetrolEngine(){

    }
    private static final String TAG = "PetrolEngine";
    @Override
    public void start() {
        Log.d(TAG, "start: Petrol engine started");
    }
}

File: AndroidStudioProjects/DaggerDemoKotlin/app/src/main/java/com/example/daggerdemokotlin/PetrolEngine.kt
package com.example.daggerdemokotlin

import javax.inject.Inject
import javax.inject.Named

class PetrolEngine : Engine {

    var powerCapacity:Int
    var engineCapacity:Int

    @Inject
    constructor(@Named("power")powerCapacity:Int, @Named("engine")engineCapacity:Int){
        this.powerCapacity= powerCapacity
        this.engineCapacity= engineCapacity
    }

    override fun start() {
        System.out.println("Petrol engine started: powerCapacity: "+powerCapacity+" engineCapacity: "+engineCapacity)
    }
}